### PR TITLE
New logging framework and lots more log lines

### PR DESCRIPTION
**Pull request by sungo**
This also alters the functionality of audit logging. All the data formally only available when auditing is now available in the regular logs. The exception is request bodies and non-error response bodies. The audit feature flag controls the inclusion of that data.

### CODE REVIEW

File: lib/Conch/Plugin/Logging.pm [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	);
+
+	$app->log($logger);
+	$app->hook(before_dispatch => sub {
+		my $c = shift;
+		$c->app->log->request_id($c->req->request_id);
+	});
+
+
+	# Render exceptions and Not Found as JSON
```
[the remark this answers is not in the document]
Comment (sungo) [reply lib/Conch/Plugin/Logging.pm]: If you mean the before_render hook there at line 59, that hook is what sends exceptions to rollbar (line 64). The other bits can certainly be extracted into a different before_render hook.

[the remark this answers is not in the document]
Comment (sungo) [reply lib/Conch/Plugin/Logging.pm]: Done.

File: lib/Conch/Plugin/Logging.pm [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+			$c->stash('user')->email . " (".$c->stash('user')->id.")" :
+			'NOT AUTHED';
+
+		my $req_headers = $c->tx->req->headers->to_hash;
+		for (qw(Authorization Cookie jwt_token jwt_sig)) {
+			delete $req_headers->{$_};
+		}
+
+		my $params = $c->req->params->to_hash;
+		if($c->req->url =~ /login/) {
```
[the remark this answers is not in the document]
Comment (sungo) [reply lib/Conch/Plugin/Logging.pm]: The special casing was pointless anyway so I removed it.

File: lib/Conch/Plugin/Logging.pm [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+	my $logger = Conch::Log->new(
+		path  => $home->child('log', "$mode.log"),
+		level => 'debug',
+	);
+
+	$app->log($logger);
+	$app->hook(before_dispatch => sub {
+		my $c = shift;
+		$c->app->log->request_id($c->req->request_id);
```
[the remark this answers is not in the document]
Comment (sungo) [reply lib/Conch/Plugin/Logging.pm]: Done.

File: lib/Conch/Controller/DB/HardwareProduct.pm [hunk 4]
```diff
@@ -1,7 +1,10 @@
 package Conch::Controller::DB::HardwareProduct;
 
+use Role::Tiny;
```
Comment (karenetheridge) [inline lib/Conch/Controller/DB/HardwareProduct.pm]: minor quibble: this should be Role::Tiny::With. (`use Role::Tiny` makes this module into a role.)  We don't want our controllers to be roles :)

Comment (sungo) [reply lib/Conch/Controller/DB/HardwareProduct.pm]: done

**Pull request merged**